crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -250,16 +441,48 @@ fn is_function_maybe_reassigned<'a>(
         })
 }
 
-// skipping whitespace, commas, finds the next character (exclusive)
+fn get_jsx_element_symbol_id<'a>(
+    node: &'a JSXElementName<'a>,
+    ctx: &'a LintContext<'_>,
+) -> Option<SymbolId> {
+    let node = match node {
+        JSXElementName::IdentifierReference(ident) => Some(ident.as_ref()),
+        JSXElementName::MemberExpression(expr) => expr.get_identifier(),
+        JSXElementName::Identifier(_)
+        | JSXElementName::NamespacedName(_)
+        | JSXElementName::ThisExpression(_) => None,
+    }?;
+
+    ctx.symbols().get_reference(node.reference_id()).symbol_id()
+}
+
+enum Direction {
+    Forward,
+    Backward,
+}
+
+// Skips whitespace and commas in a given direction and
+// returns the next character if found.
 #[allow(clippy::cast_possible_truncation)]
-fn skip_to_next_char(s: &str, start: u32) -> u32 {
-    for (i, c) in s.char_indices().skip(start as usize) {
-        if !c.is_whitespace() && c != ',' {
-            return i as u32;
-        }
+fn skip_to_next_char(s: &str, start: u32, direction: &Direction) -> Option<u32> {
+    // span is a half-open interval: [start, end)
+    // so we should return in that way.
+    let start = start.try_into().expect("converting u32 to usize should be safe");
```

Comment:
it's OK to assume the conversion will not fail (it will still panic anyway if it does)
```suggestion
    let start = start as usize;
```